mcp/streamable.go

```diff
@@ -45,6 +45,12 @@ type StreamableHTTPHandler struct {
 // StreamableHTTPOptions is a placeholder options struct for future
 // configuration of the StreamableHTTP handler.
 type StreamableHTTPOptions struct {
+	// GetSessionID provides the next session ID to use for an incoming request.
+	//
+	// If GetSessionID returns an empty string, the session is 'stateless',
+	// meaning it is not persisted and no session validation is performed.
+	GetSessionID func() string
```

Comment:
Why do we want the user to control this? (We could do statelessness with just a bool.)